flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/table.scala

```diff
@@ -251,12 +249,157 @@ class Table(
     * }}}
     */
   def join(right: Table): Table = {
+    join(right, None, JoinType.INNER)
+  }
+
+  /**
+    * Joins two [[Table]]s. Similar to an SQL join. The fields of the two joined
+    * operations must not overlap, use [[as]] to rename fields if necessary.
+    *
+    * Note: Both tables must be bound to the same [[TableEnvironment]].
+    *
+    * Example:
+    *
+    * {{{
+    *   left.join(right, "a = b && c > 3")
```

Comment:
If we decide to only allow join predicates as join condition, we should remove the `c > 3` predicate.